src/helpers/synthetic_peers.rs

```diff
@@ -0,0 +1,180 @@
+use crate::protocol::{
+    message::{constants::HEADER_LEN, filter::MessageFilter, Message},
+    payload::{codec::Codec, Version},
+};
+
+use assert_matches::assert_matches;
+use pea2pea::{
+    connections::ConnectionSide,
+    protocols::{Handshaking, Reading, Writing},
+    Connection, Node, Pea2Pea,
+};
+use tokio::sync::mpsc::{self, Receiver, Sender};
+
+use std::{
+    io::{Cursor, Result},
+    net::SocketAddr,
+};
+
+pub struct SyntheticNode {
+    inner_node: InnerNode,
+    inbound_rx: Receiver<Message>,
+}
+
+impl SyntheticNode {
+    pub fn new(node: Node, enable_handshaking: bool, message_filter: MessageFilter) -> Self {
```

Comment:
Is there a way one can re-configure this on the fly?

I'm not sure we have a use-case currently, but I'm thinking of a situtation where one wants to change `message_filter` in-between receive calls. I suppose that would also be impossible to control correctly since `Pea2Pea` starts a background task for reading the stream (?).

Reply:
We could mutate the `SyntheticNode` if this is necessary (tough we'd need to `Arc` the inner node).